cmd/abapAddonAssemblyKitReserveNextPackages.go

```diff
@@ -24,75 +24,96 @@ func abapAddonAssemblyKitReserveNextPackages(config abapAddonAssemblyKitReserveN
+	}
+	log.Entry().Info("┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┻━━━━━━━━━━━━━━━━━━━━━━┻━━━━━━━┻━━━━━━━━┻━━━━━━━━━━━━┻━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┻━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛")
+
+	log.Entry().Info("... checking and processing provided and received data")
 	addonDescriptor.Repositories, err = checkAndCopyFieldsToRepositories(packagesWithRepos)
 	if err != nil {
 		return err
 	}
 
-	log.Entry().Info("Writing package names, types, status, namespace and predecessorCommitID to CommonPipelineEnvironment")
+	log.Entry().Info("... writing AddonDescriptor (package name, type, status, namespace and predecessorCommitID) back to CommonPipelineEnvironment")
 	cpe.abap.addonDescriptor = addonDescriptor.AsJSONstring()
 	return nil
 }
 
 func checkAndCopyFieldsToRepositories(pckgWR []aakaas.PackageWithRepository) ([]abaputils.Repository, error) {
 	var repos []abaputils.Repository
-
-	log.Entry().Infof("%-30v | %-20v | %-6v | %-40v | %-40v", "Software Component", "Package", "Status", "CommitID (from addon.yml)", "PredecessorCommitID (from AAKaaS)")
-
+	var checkFailure error = nil
 	for i := range pckgWR {
-
-		log.Entry().Infof("%-30v | %-20v | %-6v | %-40v | %-40v", pckgWR[i].Repo.Name, pckgWR[i].Package.PackageName, pckgWR[i].Package.Status, pckgWR[i].Repo.CommitID, pckgWR[i].Package.PredecessorCommitID)
-
 		if !pckgWR[i].Repo.UseClassicCTS {
 			if pckgWR[i].Package.Status == aakaas.PackageStatusReleased {
 				//Ensure for Packages with Status R that CommitID of package = the one from addon.yml, beware of short commitID in addon.yml
 				addonYAMLcommitIDLength := len(pckgWR[i].Repo.CommitID)
 				if len(pckgWR[i].Package.CommitID) < addonYAMLcommitIDLength {
-					return repos, errors.New("Provided CommitIDs have wrong length: " + pckgWR[i].Repo.CommitID + "(addon.yml) longer than the one from AAKaaS " + pckgWR[i].Package.CommitID)
-				}
-				packageCommitIDsubsting := pckgWR[i].Package.CommitID[0:addonYAMLcommitIDLength]
-				if pckgWR[i].Repo.CommitID != packageCommitIDsubsting {
-					log.Entry().Error("package " + pckgWR[i].Package.PackageName + " was already build but with commit " + pckgWR[i].Package.CommitID + ", not with " + pckgWR[i].Repo.CommitID)
-					log.Entry().Error("If you want to build a new package make sure to increase the dotted-version-string in addon.yml")
-					log.Entry().Error("If you do NOT want to build a new package enter the commitID " + pckgWR[i].Package.CommitID + " for software component " + pckgWR[i].Repo.Name + " in addon.yml")
-					return repos, errors.New("commit of released package does not match with addon.yml")
+					checkFailure = errors.New("Provided CommitIDs have wrong length: " + pckgWR[i].Repo.CommitID + "(addon.yml) longer than the one from AAKaaS " + pckgWR[i].Package.CommitID)
```

Comment:
As this is inside a loop, checkFailure will only save the last error (where as the old version gives back the first error). Perhaps some kind of accumulation logic would be nice

Reply:
with the next line I write the errors to the log so they can be processed by the end user all at once (if the log is read) but the real error is only the last one...
log.Entry().WithError(checkFailure).Error(" => Check failure: to be corrected in addon.yml prior next execution")